src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -599,27 +597,33 @@ public bool Remove(TKey key)
                 int hashCode = comparer.GetHashCode(key) & 0x7FFFFFFF;
                 int bucket = hashCode % buckets.Length;
                 int last = -1;
-                for (int i = buckets[bucket]; i >= 0; last = i, i = entries[i].next)
+                int i = buckets[bucket];
+                while (i >= 0)
```

Comment:
Nope, they all look like `while` loops to JIT and it transforms them into `do`-`while` loops.

`Dictionary.Remove` diff: https://gist.github.com/mikedn/ccf5673a57ef7182c961b4e387ab5c0e